### PR TITLE
<AvatarGroup/> - new component

### PR DESCRIPTION
**Pull request by roibWix**
### 🔦 Summary

...

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6584

File: packages/wix-style-react/perfer.config.js [hunk 1]
```diff
@@ -172,6 +172,7 @@ const javascriptFiles = [
   ['VerticalTabsItem.bundle.min.js', 40],
   ['WarningIndicator.bundle.min.js', 40],
   ['WixStyleReactProvider.bundle.min.js', 10],
+  ['AvatarGroup.bundle.min.js', 57],
```
Comment (EgoziE) [inline packages/wix-style-react/perfer.config.js:175]: Sort

File: packages/wix-style-react/stories/index.js [hunk 2]
```diff
@@ -230,3 +230,6 @@ require('../src/Themes/editorX/docs');
 
 // BusinessDashboard
 require('../src/Themes/businessDashboard/docs');
+
+// TODO: move to correct position
+require('../src/AvatarGroup/docs/index.story');
```
Comment (EgoziE) [inline packages/wix-style-react/stories/index.js:235]: sort

File: packages/wix-style-react/src/index.js [hunk 3]
```diff
@@ -211,7 +211,4 @@ export {
 } from './WixStyleReactProvider';
 export { default as AnalyticsSummaryCard } from './AnalyticsSummaryCard';
 export { default as BounceAnimation } from './BounceAnimation';
-export { default as SkeletonGroup } from './SkeletonGroup';
-export { default as SkeletonRectangle } from './SkeletonRectangle';
-export { default as SkeletonCircle } from './SkeletonCircle';
-export { default as SkeletonLine } from './SkeletonLine';
```
Comment (EgoziE) [inline packages/wix-style-react/src/index.js:217]: Don't delete other components! 😂

File: packages/wix-style-react/src/AvatarGroup/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  dataHook?: string;
+  className?: string;
+  type?: GroupType;
+  size?: AvatarGroupSize;
+  showDivider?: boolean;
+  maxItems?: number;
+  items: CommonAvatarProps[];
+}
+export type GroupType = 'stretched' | 'condensed';
+export type AvatarGroupSize = 'medium' | 'small';
```
Comment (EgoziE) [inline packages/wix-style-react/src/AvatarGroup/index.d.ts:13]: Move up, so GroupType is first defined, then used.

File: packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    /**
+     * Gets the number Of displayed Avatars
+     * @returns {Promise<number>}
+     */
+    getAvatarsCount: () => countByHook(base, dataHooks.avatarGroupItem),
+
+    /**
+     * Gets the text content
+     * @param {number} Avatar index
```
Comment (EgoziE) [inline packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.js:21]: The pattern is:
@param {<type>} <name> <description>


     * @param {number} index Avatar index

File: packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+     * @returns {Promise<string>}
+     */
+    getAvatarContentByIndex: async n =>
```
Comment (EgoziE) [inline packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.js:24]: Change "n" to "index"

File: packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.d.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import { BaseUniDriver } from 'wix-ui-test-utils/unidriver';
+
+export interface AvatarGroupUniDriver extends BaseUniDriver {
+  getAvatarsCount(): Promise<number>;
+  getAvatarContentByIndex(times: number): Promise<string>;
```
Comment (EgoziE) [inline packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.d.ts:5]: change "times" to "index"

File: packages/wix-style-react/src/AvatarGroup/AvatarGroup.st.css [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  color: value(D20);
+  border-radius: 50%;
+  user-select: none;
+  z-index: 1;
+}
+
+
+.moreItemAvatar :global([data-hook*='avatar-core']) > div {
+  color: value(D20) !important;
+}
```
Comment (EgoziE) [inline packages/wix-style-react/src/AvatarGroup/AvatarGroup.st.css:30]: Let's talk about extending css in Stylable!
Please schedule 30 min meeting next week.

**Review by EgoziE — CHANGES_REQUESTED** [review]

File: packages/wix-style-react/src/AvatarGroup/AvatarGroup.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+   * Show divider on the side of the first Avatar
+   */
+  divider: PropTypes.bool,
+
+  /**
+   * Sets the number of Avatars to display
+   */
+  maxAvatarLimit: PropTypes.number,
+
+  avatars: PropTypes.arrayOf(Avatar),
```
Comment (shlomitc) [inline packages/wix-style-react/src/AvatarGroup/AvatarGroup.js:109]: looks redundant

**Review by milkyfruit — CHANGES_REQUESTED** [review]
@roibWix I noticed two UI issues:

- [ ] More number indicator is missing when size is small
image

- [ ] Divider is missing 12px margin from the right side
image

**Comment by roibWix** [comment]
@milkyfruit,
Where did you see the margin-right issue?

**Comment by milkyfruit** [comment]
> @milkyfruit,
> Where did you see the margin-right issue?

It doesn't show up anymore, it looked fixed.

**Review by milkyfruit — APPROVED** [review]

**Comment by shlomitc** [comment]
@roibWix @EgoziE 
I'm not entirely sure regarding the `moreItemContet` prop
1. it's not documented
2. doc doesn't look good
image
3. I have some concerns regarding its usage. Seems like it is always a popover, but is this the only case? Can it be a simple button click?
I remember there was a request for a dropdown there. Obviously, you can compose `DropdownLayout`, but then not all a11y features are met. Maybe it should be a bit more open to composition?

**Comment by justinara** [comment]
@shlomitc @roibWix i double checked it with @ayeletr who defined this component - primary idea is that more indication can be clickable, but popover is not a default and single option it opens. It could open, let's say, a modal instead, or anything else basically.  I think it should behave little more like a button, with onClick handler, etc.

**Comment by shlomitc** [comment]
Thanks @justinara !
In that case we probably need to do some API adjustments to become more composable

**Comment by justinara** [comment]
@roibWix this is a doc, which includes component description, updated examples with descriptions, also improved prop descriptions for API section: https://docs.google.com/document/d/1seHSUTej2WKqY2Ar6mX-reuITXyTzY8up-u9nPHoRlc/edit?usp=sharing

@shlomitc fyi

**Review by EgoziE — CHANGES_REQUESTED** [review]

File: packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+     * @returns {Promise<boolean>}
+     */
+    isMoreIndicatorExist: async () =>
+      await findByHook(base, dataHooks.avatarGroupMoreItem).exists(),
+
+    /**
+     * Hovers the component
+     * @returns {Promise<void>}
+     */
+    hover: async () => await findByHookAtIndex(base, dataHooks.avatarGroupItem).hover()
```
[the remark this answers is not in the document]
Comment (roibWix) [reply packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.js:57]: On second thought, it's unnecessary.

File: packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    getAvatarTextContentByIndex: async index =>
+      await findByHookAtIndex(base, dataHooks.avatarGroupItem, index).text(),
+
+    /**
+     * Gets the content type
+     * @param {number} index Avatar index
+     * @returns {Promise<string>}
+     */
+    getAvatarContentTypeByIndex: async index => {
```
[the remark this answers is not in the document]
Comment (roibWix) [reply packages/wix-style-react/src/AvatarGroup/AvatarGroup.uni.driver.js:33]: same here, unnecessary

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-ui-tpa**

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_6584/

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-style-react**

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6584

File: packages/wix-style-react/src/AvatarGroup/AvatarGroup.st.css [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+  border: 2px solid white;
+}
+
+/* moreItem */
+
+.moreItemContainer {
+  border-radius: 50%;
+  background-color: white;
+  border: 2px solid value(D10-10);
+  -st-states: size(enum(small, medium)), type(enum(stretched, condensed));
```
Comment (EgoziE) [inline packages/wix-style-react/src/AvatarGroup/AvatarGroup.st.css:37]: states should be first

**Review by EgoziE — CHANGES_REQUESTED** [review]

**Review by EgoziE — APPROVED** [review]

**Review by justinara — CHANGES_REQUESTED** [review]
Well done! A few tiny requests: 
1. Use **Layout** instead of **Box** to layout component in separate rows in size and group type examples. As per doc here: https://docs.google.com/document/d/1seHSUTej2WKqY2Ar6mX-reuITXyTzY8up-u9nPHoRlc/edit?usp=sharing
2. I've updated each example description a little bit - could we update them in the story too? Same doc: https://docs.google.com/document/d/1seHSUTej2WKqY2Ar6mX-reuITXyTzY8up-u9nPHoRlc/edit?usp=sharing
3. Divider should have an equal 12px distance from avatars on both sides - left and right. It seems to be a little closer to the list on the right at the moment:
image zeplin: https://zpl.io/VD0DzxJ

**Comment by TalZa1810** [comment]
Hi @roibWix ,
Can you please add this component to the cheatsheet ?
You can do it in a separate PR if you prefer (after merging). 
You can find the instructions how to do it [here](https://github.com/wix/wix-style-react/blob/master/packages/wix-style-react/docs/contribution/CHEATSHEET_GUIDLINES.md).

Feel free to ask me questions if you have any.

Thanks.

**Review by justinara — APPROVED** [review]
Description for 'Size' example is not up to date

**Pull request merged**